src/main/java/seedu/address/model/todolist/TaskName.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.todolist;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Task's name in the todo list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTaskName(String)}
+ */
+public class TaskName {
+    public static final String MESSAGE_CONSTRAINTS = "Task name should be no longer than 30 words.";
+
+    private final String value;
+
+    /**
+     * Constructs a {@code TaskName}.
+     *
+     * @param taskName valid task name.
+     */
+    public TaskName(String taskName) {
+        requireNonNull(taskName);
+        this.value = taskName;
+    }
+
+    /**
+     * Returns true if a given string is a valid task name.
+     *
+     * @param test given string
+     * @return true if task name is valid
+     */
+    public static boolean isValidTaskName(String test) {
```

Comment:
Perhaps this validation test could be modified to account for empty task name as well?

Reply:
Yup, totally forgot about that 😆